src/app/options/qgsoptions.cpp

```diff
@@ -1161,19 +1161,10 @@ QgsOptions::QgsOptions( QWidget *parent, Qt::WindowFlags fl, const QList<QgsOpti
   //default snap mode
   mSnappingEnabledDefault->setChecked( QgsSettingsRegistryCore::settingsDigitizingDefaultSnapEnabled.value() );
 
-  for ( Qgis::SnappingType type :
-        {
-          Qgis::SnappingType::NoSnap,
-          Qgis::SnappingType::Vertex,
-          Qgis::SnappingType::Segment,
-          Qgis::SnappingType::Centroid,
-          Qgis::SnappingType::MiddleOfSegment,
-          Qgis::SnappingType::LineEndpoint,
-          Qgis::SnappingType::Area,
-        } )
-  {
-    mDefaultSnapTypeComboBox->addItem( QgsSnappingConfig::snappingTypeFlagToIcon( type ),
-                                       QgsSnappingConfig::snappingTypeFlagToString( type ),
+  for ( Qgis::SnappingType type : qgsEnumMap<Qgis::SnappingType>().keys() )
```

Comment:
This looks rather inefficient -- we build a map of the key to values, then throw away the values and build another list of just the keys. This calls for a more appropriate convience method to just return a list/vector of the keys directly, without the multiple container constructions and iterations.

Reply:
I just added a new method which does this [f2243e1](https://github.com/qgis/QGIS/pull/47535/commits/f2243e18b8c7a35156d2e7012bbcfbda8a4b3e4b)